code/filesystem/locator/factory.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ * Kodekit - http://timble.net/kodekit
+ *
+ * @copyright   Copyright (C) 2007 - 2016 Johan Janssens and Timble CVBA. (http://www.timble.net)
+ * @license     MPL v2.0 <https://www.mozilla.org/en-US/MPL/2.0>
+ * @link        https://github.com/timble/kodekit for the canonical source repository
+ */
+
+namespace Kodekit\Library;
+
+/**
+ * Filesystem Locator Factory
+ *
+ * @author  Johan Janssens <http://github.com/johanjanssens>
+ * @package Kodekit\Library\Filesystem\Locator
+ */
+class FilesystemLocatorFactory extends Object implements ObjectSingleton
+{
+    /**
+     * Registered locators
+     *
+     * @var array
+     */
+    private $__locators;
+
+    /**
+     * Constructor.
+     *
+     * @param ObjectConfig $config Configuration options
+     */
+    public function __construct( ObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        //Auto register locators
+        foreach($config->locators as $locator) {
+            $this->registerLocator($locator);
+        }
+    }
+
+    /**
+     * Initializes the options for the object
+     *
+     * Called from {@link __construct()} as a first step of object instantiation.
+     *
+     * @param   ObjectConfig $config Configuration options.
+     * @return  void
+     */
+    protected function _initialize(ObjectConfig $config)
+    {
+        $config->append(array(
+            'locators' => array(
+                'lib:filesystem.locator.file'
+            ),
+        ));
+
+        parent::_initialize($config);
+    }
+
+    /**
+     * Create a locator
+     *
+     * Note that only URLs delimited by "://"" and ":" are supported, ":/" while technically valid URLs, are not.
+     * If no locator is registered for the specific scheme a exception will be thrown.
+     *
+     * @param  string $path    The locator path
+     * @param  array  $config  An optional associative array of configuration options
+     * @throws \InvalidArgumentException If the path is not valid
+     * @throws \RuntimeException         If the locator isn't registered
+     * @throws \UnexpectedValueException If the locator object doesn't implement the FilesystemLocatorInterface
+     * @return FilesystemLocatorInterface
+     */
+    public function createLocator($path, array $config = array())
+    {
+        $scheme = parse_url($path, PHP_URL_SCHEME);
+
+        //If no scheme is specified fall back to file:// locator
+        $name = !empty($scheme) ? $scheme : 'file';
+
+        //If a windows drive letter is passed use file:// locator
+        if(strtoupper(substr(PHP_OS, 0, 3)) === 'WIN')
+        {
+            if(preg_match('#^[a-z]{1}$#i', $name)) {
+                $name = 'file';
+            }
+        }
+
+        //Locator not supported
+        if(!in_array($name, $this->getLocators()))
```

Comment:
Should this not be an `isset()`? Looks like `$this-> __locators` is an array indexed by name